### PR TITLE
[choreo] Event marker refactor

### PR DESCRIPTION
**Pull request by shueja**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,4 +2,100 @@
     "rust-analyzer.check.overrideCommand": [
         "cargo", "clippy", "--target-dir", "./target/clippy"
     ],
+    "files.associations": {
```
Comment (spacey-sooty) [inline .vscode/settings.json:5]: These should probably be in the user settings.json

**Comment by shueja** [comment]
C++, Java and Python deserialization changes have been tested. Ready for review.

**Review by calcmogul — APPROVED** [review]
Does the event marker stuff have C++ tests?

**Review by calcmogul — CHANGES_REQUESTED** [review]
Should add an assert to the Java test while you're at it. The commented out imports in ChoreoTests.java can also be removed.

**Review by calcmogul — APPROVED** [review]

**Pull request merged**